src/components/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -46,37 +46,33 @@ export const Breadcrumbs = ({
 
   const ref = React.useRef<HTMLUListElement>(null);
 
-  /**
-   * Checks if breadcrumbs would overflow
-   */
-  const updateShouldTruncate = () => {
-    setShouldTruncate(false);
-    if (
-      ref &&
-      ref.current &&
-      ref.current.clientWidth < ref.current.scrollWidth
-    ) {
-      setShouldTruncate(true);
-    }
-  };
-  const debouncedUpdateShouldTruncate = debounce(updateShouldTruncate, 200);
-
   /**
    * Needs useLayoutEffect over useEffect since it needs to be run before paint.
    * TODO: with React 18, might be able to use useEffect https://github.com/reactjs/reactjs.org/blob/d14cbdca2445cd676526c4c52e1e106342ff7bb3/content/docs/hooks-reference.md?plain=1#L155
    */
   React.useLayoutEffect(() => {
+    const updateShouldTruncate = () => {
+      const willOverflow = ref.current
+        ? ref.current.clientWidth < ref.current.scrollWidth
+        : false;
+
+      setShouldTruncate(willOverflow);
```

Comment:
Note the change of logic from

```ts
setShouldTruncate(false);
if (someConditions) {
  setShouldTruncate(true);
}
```

to

```ts
setShouldTruncate(someConditions);
```

Result should be the same, since `setState` updates get batched. Everything seems the same in storybook and in the added unit tests, too.